packages/devtools/devtools-testbench/src/App.tsx

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import React, { FC, useEffect, useState } from 'react';
+import { HashRouter } from 'react-router-dom';
+
+import { Client, fromHost } from '@dxos/client';
+import { ClientAndServices, PanelsContainer, sections } from '@dxos/devtools';
+import { AppView, OptionsContext, Routes as KaiRoutes } from '@dxos/kai';
+import { Config, Defaults } from '@dxos/config';
+import { ClientContext } from '@dxos/react-client';
+
+/**
+ * Main app container with routes.
+ */
+export const App: FC<{ views: AppView[]; debug?: boolean }> = ({ views, debug = false, demo = true }) => {
+  const [value, setValue] = useState<ClientAndServices | undefined>(undefined);
+
+  // Auto-create client and profile.
+  useEffect(() => {
+    setTimeout(async () => {
+      const config = new Config(Defaults());
+      const services = fromHost(config);
+      const client = new Client({
+        config,
+        services
+      });
+
+      await client.initialize();
+      // TODO(burdon): Hangs (no error) if profile not created?
+      if (!client.halo.profile) {
+        await client.halo.createProfile();
+      }
+
+      setValue({ client, services: services.services });
+    });
+  }, []);
```

Comment:
I don't love this but understand why it's being done this way given how ClientProvider currently works. Let's circle back to it later.

Reply:
Sure